File: spec/functional_test/fixtures/java/play/app/controllers/MissingController.java
package controllers;

import play.mvc.Controller;
import play.mvc.Result;

public class MissingController extends Controller {

    public Result multipart() {
        // Should be form/multipart
        Object body = request().body().asMultipartFormData();
        return ok("multipart");
    }

    public Result bytes() {
        // Should be body/bytes
        byte[] b = request().body().asBytes();
        return ok("bytes");
    }

    public Result whitespace() {
        // Should be json
        request() . body() . asJson();
        return ok("json");
    }
}

File: spec/functional_test/fixtures/scala/play/app/controllers/MissingController.scala
package controllers

import play.api.mvc._

class MissingController extends BaseController {
  def multipart() = Action { request =>
    request.body.asMultipartFormData
    Ok("multipart")
  }

  def xml() = Action { request =>
    request.body.asXml
    Ok("xml")
  }

  def whitespace() = Action { request =>
    request . body . asJson
    Ok("json")
  }
}
